server/consumer.go

```diff
@@ -664,28 +664,13 @@ func (o *Consumer) updateDeliverSubject(newDeliver string) {
 	o.mu.Lock()
 	defer o.mu.Unlock()
 
-	mset := o.mset
-	if mset == nil || o.isPullMode() {
+	if o.mset == nil || o.isPullMode() {
```

Comment:
Could you use `o.closed` here instead of `o.mset`? I fixed recently the race on close (due to release of consumer's lock when sending advisory).

Reply:
yes will fix.